src/dsp_tools/utils/excel_to_json_lists.py

```diff
@@ -34,20 +34,21 @@ def expand_lists_from_excel(
 
     Args:
         lists_section: the "lists" section of a parsed JSON project file. If this is an empty list, an empty list will be returned.
+    
+    Raises:
+        BaseError: if a problem occurred while trying to expand the Excel files
 
     Returns:
         the same "lists" section, but without references to Excel files
-        True if all lists could be expanded correctly, False if a problem occurred
     """
-    overall_success = True
     new_lists = []
     for _list in lists_section:
         if "folder" not in _list["nodes"]:
             # this list is a JSON list: return it as it is
             new_lists.append(_list)
         else:
             # this is a reference to a folder with Excel files
-            foldername = _list["nodes"]["folder"]
+            foldername = _list["nodes"]["folder"]   # type: ignore
```

Comment:
Thanks for the idea, but unfortunately the error lies on a deeper level.

Reply:
too bad... maybe it would be interesting to model the whole domain entities as objects (dataclass, pydantic, ...) instead of using dictionaries everywhere. That would be a lot of work, but I think it would improve both development experience as well as code quality